### PR TITLE
[FLINK-23556][tests] Make SQLClientSchemaRegistryITCase more stable

### PR DESCRIPTION
**Pull request by bgeng777**
## What is the purpose of the change

This PR aims at improving the pass rate of SQLClientSchemaRegistryITCase(an unstable e2e test case).

## Brief change log

  - Remove 120 seconds limitation of the test. Due to the complexity of this e2e case, the old time limitation will make the case fail easily. Besides, after removing the limit, if this test fails, we can get its meaningful logs instead of a single timeout error.
  - Use ClassRule instead of Rule to reuse created containers which will save resources and time.
  - Increate max waiting time of some functions to make it easier to success.

## Verifying this change
- This change is already covered by existing tests, such as SQLClientSchemaRegistryITCase.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers:  no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper:  no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/src/test/java/org/apache/flink/tests/util/kafka/SQLClientSchemaRegistryITCase.java [hunk 1]
```diff
@@ -60,25 +62,32 @@
...
-                    .withNetworkAliases(INTER_CONTAINER_KAFKA_ALIAS);
+    @ClassRule
+    private static final KafkaContainer kafka =
+            new KafkaContainer(DockerImageName.parse(DockerImageVersions.KAFKA)) {
+                @Override
+                protected void doStart() {
+                    super.doStart();
+                    this.followOutput(LOG_CONSUMER);
+                }
+            }.withNetwork(network).withNetworkAliases(INTER_CONTAINER_KAFKA_ALIAS);
```
Comment (wuchong) [inline flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/src/test/java/org/apache/flink/tests/util/kafka/SQLClientSchemaRegistryITCase.java:87]: Can we use `.setLogConsumers()` to set the `LOG_CONSUMER`? It seems overriding `doStart` is a little trick to me.

Comment (bgeng777) [reply flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/src/test/java/org/apache/flink/tests/util/kafka/SQLClientSchemaRegistryITCase.java]: Hi wuchong, thanks a lot for your review. I have changed the overriding to `withLogConsumer()` for readability. 
Let me know if it is better.

**Review by wuchong — APPROVED** [review]
LGTM.

**Comment by bgeng777** [comment]
@flinkbot run azure

**Comment by bgeng777** [comment]
hi @wuchong , the pr has passed the azure ci. Would you mind merging it into the master branch? If there are any other concerns, please let me know, thanks~

**Comment by wuchong** [comment]
Sure

**Pull request merged**